File: app/src/main/java/ru/willdes/nginxplus/ServersAdapter.java
package ru.willdes.nginxplus;

import android.annotation.SuppressLint;
import android.content.Context;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class ServersAdapter extends RecyclerView.Adapter<ServersAdapter.ServersViewHolder> {

    private static final String TAG = "RecyclerViewAdapter";

    private List<ServersModel> list = new ArrayList<>();
    private Context context;

    public ServersAdapter(Context context, List<ServersModel> list) {
        this.list = list;
        this.context = context;
    }

    @NonNull
    @Override
    public ServersViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_servers, parent, false);
        return new ServersViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull final ServersViewHolder serversViewHolder, final int i) {
        final ServersModel temp = this.list.get(i);
        serversViewHolder.srvname.setText(temp.getName());
        serversViewHolder.twActive.setText("" + temp.getActive());
        serversViewHolder.twRequests.setText("" + temp.getRequests());
        serversViewHolder.twActive.setBackgroundResource(android.R.color.transparent);
        serversViewHolder.twRequests.setBackgroundResource(android.R.color.transparent);

        switch (temp.getState()) {
            case "up":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorUp));
                if (temp.getActive() >= 2 && temp.getRequests() == 0) {
                    //Log.d("Orange", "name: " + temp.getName() + "; id: " + i + " set orange color" );
                    serversViewHolder.twActive.setBackgroundResource(R.color.orange);
                    serversViewHolder.twRequests.setBackgroundResource(R.color.orange);
                }
                break;

            case "down":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorDown));
                break;

            case "unavail":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorRemove));
                break;

            case "draining":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorDrain));
                break;

            case "checking":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorUp));
                break;

            case "unhealthy":
                serversViewHolder.ivStat.setColorFilter(context.getResources().getColor(R.color.colorRemove));
                break;

        }

        serversViewHolder.ibUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Runnable runnable = new Runnable() {
                    @Override
                    public void run() {
                        new POSTtoServer(temp.getId(), temp.getName(), "up");
                    }
                };
                Thread thread = new Thread(runnable);
                thread.start();
            }
        });
        serversViewHolder.ibDown.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Runnable runnable = new Runnable() {
                    @Override
                    public void run() {
                        new POSTtoServer(temp.getId(), temp.getName(), "down");
                    }
                };
                Thread thread = new Thread(runnable);
                thread.start();
            }
        });
        serversViewHolder.ibDrain.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Runnable runnable = new Runnable() {
                    @Override
                    public void run() {
                        new POSTtoServer(temp.getId(), temp.getName(), "drain");
                    }
                };
                Thread thread = new Thread(runnable);
                thread.start();
            }
        });
    }

    @Override
    public int getItemCount() {
        return this.list.size();
    }

    public class ServersViewHolder extends RecyclerView.ViewHolder {
        private ImageView ivStat;
        private TextView srvname;
        private TextView twActive;
        private TextView twRequests;
        private ImageButton ibUp;
        private ImageButton ibDown;
        private ImageButton ibDrain;


        public ServersViewHolder(@NonNull View itemView) {
            super(itemView);
            ivStat = itemView.findViewById(R.id.ivStat);
            srvname = itemView.findViewById(R.id.srvname);
            twActive = itemView.findViewById(R.id.twActive);
            twRequests = itemView.findViewById(R.id.twRequests);
            ibUp = itemView.findViewById(R.id.ibUp);
            ibDown = itemView.findViewById(R.id.ibDown);
            ibDrain = itemView.findViewById(R.id.ibDrain);

        }
    }
}

File: app/src/main/java/ru/willdes/nginxplus/ServersAdapter.kt
package ru.willdes.nginxplus

import android.annotation.SuppressLint
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import ru.willdes.nginxplus.ServersAdapter.ServersViewHolder
import java.util.*

class ServersAdapter(context: Context, list: List<ServersModel>) : RecyclerView.Adapter<ServersViewHolder>() {
    private var list: List<ServersModel> = ArrayList()
    private val context: Context
    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ServersViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_servers, parent, false)
        return ServersViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(serversViewHolder: ServersViewHolder, i: Int) {
        //Log.d(TAG, "onBindViewHolder: called.");
        val temp = list[i]
        serversViewHolder.srvname.text = "       " + temp.name
        serversViewHolder.twActive.text = "" + temp.active
        serversViewHolder.twRequests.text = "" + temp.requests
        serversViewHolder.twActive.setBackgroundResource(android.R.color.transparent)
        serversViewHolder.twRequests.setBackgroundResource(android.R.color.transparent)
        when (temp.state) {
            "up" -> {
                serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorUp))
                if (temp.active >= 2 && temp.requests == 0) {
                    Log.d("Orange", "name: " + temp.name + "; id: " + i + " set orange color")
                    serversViewHolder.twActive.setBackgroundResource(R.color.orange)
                    serversViewHolder.twRequests.setBackgroundResource(R.color.orange)
                }
            }
            "down" -> serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorDown))
            "unavail" -> serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorRemove))
            "draining" -> serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorDrain))
            "checking" -> serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorUp))
            "unhealthy" -> serversViewHolder.ivStat.setColorFilter(context.resources.getColor(R.color.colorRemove))
        }
        serversViewHolder.ibUp.setOnClickListener {
            val runnable = Runnable { POSTtoServer(i, temp.name, "up") }
            val thread = Thread(runnable)
            thread.start()
        }
        serversViewHolder.ibDown.setOnClickListener {
            val runnable = Runnable { POSTtoServer(i, temp.name, "down") }
            val thread = Thread(runnable)
            thread.start()
        }
        serversViewHolder.ibDrain.setOnClickListener {
            val runnable = Runnable { POSTtoServer(i, temp.name, "drain") }
            val thread = Thread(runnable)
            thread.start()
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ServersViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal val ivStat: ImageView
        internal val srvname: TextView
        internal val twActive: TextView
        internal val twRequests: TextView
        internal val ibUp: ImageButton
        internal val ibDown: ImageButton
        internal val ibDrain: ImageButton

        init {
            ivStat = itemView.findViewById(R.id.ivStat)
            srvname = itemView.findViewById(R.id.srvname)
            twActive = itemView.findViewById(R.id.twActive)
            twRequests = itemView.findViewById(R.id.twRequests)
            ibUp = itemView.findViewById(R.id.ibUp)
            ibDown = itemView.findViewById(R.id.ibDown)
            ibDrain = itemView.findViewById(R.id.ibDrain)
        }
    }

    companion object {
        private const val TAG = "RecyclerViewAdapter"
    }

    init {
        this.list = list
        this.context = context
    }
}